massa-protocol-worker/src/protocol_network.rs

```diff
@@ -353,6 +353,13 @@ impl ProtocolWorker {
 
         // Check operation_list against expected operations hash from header.
         if header.content.operation_merkle_root == Hash::compute_from(&total_hash) {
+
+            if operation_ids.len() > self.config.max_operations_per_block as usize {
+                warn!("Node id {} sent us a operation list for block id {} but the operations we already have in our records exceed max operations per block constant.", from_node_id, block_id);
```

Comment:
it's not the operations we have in our records, but the list of operation_ids that were sent